src/metabase/lib/field.cljc

```diff
@@ -787,16 +787,18 @@
 (mu/defn ^:private remapped-field :- [:maybe ::lib.schema.metadata/column]
   [metadata-providerable :- ::lib.schema.metadata/metadata-providerable
    column                :- ::lib.schema.metadata/column]
-  (when-let [remap-field-id (get-in column [:lib/external-remap :field-id])]
-    (lib.metadata/field metadata-providerable remap-field-id)))
+  (or
+   (when (lib.types.isa/primary-key? column)
```

Comment:
Please note that this won't work for models for at least 2 reasons:
- We only load model metadata itself and not raw tables, that's why `lib.metadata/field` won't return the field
- Model could have metadata overrides and we should search `type/Name` within the model itself.

Reply:
@bshepherdson